src/plugins/ui_actions/public/cell_actions/components/cell_action_item.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo } from 'react';
+
+import { EuiButtonIcon, EuiToolTip, IconType } from '@elastic/eui';
+import type { Action } from '../../actions';
+import { CellActionExecutionContext } from './cell_actions';
+
+export const ActionItem = ({
+  action,
+  actionContext,
+  showTooltip,
+}: {
+  action: Action;
+  actionContext: CellActionExecutionContext;
+  showTooltip: boolean;
+}) => {
+  const actionProps = useMemo(
+    () => ({
+      iconType: action.getIconType(actionContext) as IconType,
```

Comment:
would it make sense to update the `Action` type to make `getIconType` return `IconType` instead of `string | undefined`

Reply:
Using `IconType`instead of `string` makes sense to me. I think it is a `string` because `EuiContextMenuItemIcon` expects a `string` instead of `IconType`. So updating it might lead to cascading changes that could grow big.

Tagging @Dosant and @vadimkibana to collect their input.